### ISSUE TITLE
upspinserver: unknown storage backend type "Disk"

### ISSUE THREAD

**Issue by Merovius** [upspin/upspin#544]
**What version of Upspin are you using?**

On the server, I'm using the docker image from gcr.io/upspin-containers/upspinserver:latest, built at commit 
f3dcb57e4f3923e463739246bc591853ce569b74
On the client, I'm using commit bd8a94e8b649dbe82a15136263b1824de376d955

**What operating system and processor architecture are you using?**

On the server:
Linux jeremy 4.13.9-coreos #1 SMP Thu Oct 26 03:21:00 UTC 2017 x86_64 AMD Athlon 5370 APU with Radeon R3 AuthenticAMD GNU/Linux
On the client:
Linux hix 4.13.12-200.fc26.x86_64 #1 SMP Wed Nov 8 16:47:26 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

**What did you do?**

I tried following https://upspin.io/doc/signup.md, leading me to https://upspin.io/doc/server_setup.md
I want to run both the directory server and the storage server on my NAS at home. As I'm using CoreOS, I am using the upspinserver docker image. I successfully set up a user account and the server is running; when I direct my browser to it, I get an "unconfigured upspin server", I got to [configure upspinserver](https://upspin.io/doc/server_setup.md#configure):

mero@hix ~$ master upspin setupserver -domain=merovius.de -host=jeremy.merovius.de

(note: I've shut down the upspin server for now, as I don't want anyone to write their own config sneakily :) )

**What did you expect to see?**

> Successfully put "upspin@example.com" to the key server.
> Configured upspinserver at "upspin.example.com:443".
> Created root "you@gmail.com".

**What did you see instead?**

> User "upspin@merovius.de" already exists on key server.
> upspin: setupserver: upspinserver returned status 500 Internal Server Error:
> store/server.New: invalid operation:
> 	cloud/storage.Dial: unknown storage backend type "Disk"

**Comment by Merovius** [comment]
I built the upspinserver binary from HEAD (b6cbee6f6dfe62ee7474db0dc25e0ed106cd4969), scp'ed it to /tmp and ran it with
/tmp/upspinserver -addr jeremy.merovius.de:443 -https :443 -web -letscache /srv/upspin/letscache -serverconfig /srv/upspin/server/ -config /srv/upspin/server/serverconfig.json
With that, I can successfully setup the server.

I also noticed, that the commit from the version output in the container doesn't seem to exist in this repo:
Nov 25 16:32:04 jeremy docker[1382]: 2017/11/25 16:32:04.166552 upspinserver built on Nov 22 01:50:57 UTC at commit f3dcb57e4f3923e46373

**Comment by adg** [comment]
`gcr.io/upspin-containers/upspinserver` is misnamed. It should really be `upspinserver-gcp`. It's the version from the gcp.upspin.io repository. I'm sorry for the confusion.

Really we should have multiple Docker images; one for each storage backend. I'll rename this issue to be about that, if you agree.

**Comment by adg** [comment]
> I also noticed, that the commit from the version output in the container doesn't seem to exist in this repo:

That's the commit from the gcp repo. I can see how this might confuse.

**Comment by Merovius** [comment]
Yes, I'm fine with the rename. In the meantime, I built my own Dockerfile to deploy and am using that. But eventually, having a working one from upstream that gets auto-updated would be preferable of course :)
